otter_welcome_buddy/cogs/hiring_timelines.py

```diff
@@ -1,59 +1,77 @@
+from discord import TextChannel
 from discord.ext import commands
+from discord.ext.commands import Bot
+from discord.ext.commands import Context
 
 from otter_welcome_buddy.common.constants import CronExpressions
 from otter_welcome_buddy.common.utils.dates import DateUtils
+from otter_welcome_buddy.common.utils.types.common import DiscordChannelType
 from otter_welcome_buddy.formatters import timeline
 from otter_welcome_buddy.log import wrapper
 
 
 class Timelines(commands.Cog):
     """Hiring events for every month"""
 
-    def __init__(self, bot, messages_formatter):
-        self.bot = bot
-        self.messages_formatter = messages_formatter
+    def __init__(self, bot: Bot, messages_formatter: type[timeline.Formatter]):
+        self.bot: Bot = bot
+        self.messages_formatter: type[timeline.Formatter] = messages_formatter
         self.scheduler: AsyncIOScheduler = AsyncIOScheduler()
+
         self.__configure_scheduler()
 
-    # Commented the !start !stop since I always forget to turn on the trigger
-    # @commands.command()
-    # async def start(self, _):
-    # """Command to interact with the bot and start cron"""
-    # self.__configure_scheduler()
+    @commands.group(
+        brief="Commands related to Timelines messages!",
+        invoke_without_command=True,
+        pass_context=True,
+    )
+    async def timelines(self, ctx: Context) -> None:
+        """
+        Timelines will help to keep track of important events to be announced via cronjobs
+        """
+        await ctx.send_help(ctx.command)
+
+    @timelines.command(brief="Start the cronjob for the timeline messages")  # type: ignore
+    async def start(self, _: Context) -> None:
+        """Command to interact with the bot and start cron"""
+        self.__configure_scheduler()
 
-    # @commands.command()
-    # async def stop(self, _):
-    # """Command to interact with the bot and start cron"""
-    # self.scheduler.stop()
+    @timelines.command(brief="Stop the cronjob for the timeline messages")  # type: ignore
+    async def stop(self, _: Context) -> None:
+        """Command to interact with the bot and stop cron"""
+        self.scheduler.stop()
 
     @wrapper.log_function()
-    def __configure_scheduler(self):
+    def __configure_scheduler(self) -> None:
         """Configure and start scheduler"""
         self.scheduler.add_job(
             self.send_message_on_channel,
             DateUtils.create_cron_trigger_from(
-                CronExpressions.DAY_ONE_OF_EACH_MONTH_CRON.value
+                CronExpressions.DAY_ONE_OF_EACH_MONTH_CRON.value,
             ),
         )
         self.scheduler.start()
 
-    def _get_hiring_events(self):
+    def _get_hiring_events(self) -> str:
         """Get hiring events for current month"""
         return self.messages_formatter.get_hiring_events_for(
-            DateUtils.get_current_month()
+            DateUtils.get_current_month(),
         )
 
     @wrapper.log_function()
```

Comment:
We need to do a merge from main to avoid overriding this changes

Reply:
Done with the merge